Client/Frontend/Browser/TabManager.swift

```diff
@@ -766,9 +766,14 @@ class TabManager: NSObject {
         imageStore?.clearExcluding(savedUUIDs)
     }
 
-    fileprivate lazy var restoreTabsInternal: () -> Tab? = {
+    fileprivate static var restoreOnceToken = false
+    fileprivate func restoreTabsInternal() -> Tab? {
+        assert(Thread.isMainThread)
+        guard !TabManager.restoreOnceToken else { return nil }
```

Comment:
Double negative here is kind of weird, just use `if`:

```suggestion
        if TabManager.restoreOnceToken { return nil }
```